langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/statements/foreach/ForeachTupleTests.java

```diff
@@ -129,14 +130,36 @@ public void testTupleWithNullElements() {
         Assert.assertEquals(returns[0].stringValue(), "0:d0 1: 2:d2 3: ");
     }
 
+    @Test(dataProvider = "dataToTestTupleWithRestDescriptorInForeach")
+    public void testTupleWithRestDescriptorInForeach(String functionName) {
+        BRunUtil.invoke(program, functionName);
+    }
+
+    @DataProvider
+    public Object[] dataToTestTupleWithRestDescriptorInForeach() {
+        return new Object[]{
+                "testTupleWithRestDescriptorInForeach1",
+                "testTupleWithRestDescriptorInForeach2",
+                "testTupleWithRestDescriptorInForeach3",
+                "testTupleWithRestDescriptorInForeach4",
+                "testTupleWithRestDescriptorInForeach5"
+        };
+    }
+
     @Test
     public void testNegativeTupleForeach() {
         negative = BCompileUtil.compile("test-src/statements/foreach/foreach_tuples_negative.bal");
-        Assert.assertEquals(negative.getErrorCount(), 3);
+        Assert.assertEquals(negative.getErrorCount(), 6);
         int i = 0;
         BAssertUtil.validateError(negative, i++, "incompatible types: expected '(int|string)', found 'string'", 20, 13);
         BAssertUtil.validateError(negative, i++, "incompatible types: expected '(int|string)', found 'int'", 23, 13);
-        BAssertUtil.validateError(negative, i,
+        BAssertUtil.validateError(negative, i++,
                 "incompatible types: expected '(int|string|boolean)', found '(string|int)'", 31, 13);
+        BAssertUtil.validateError(negative, i++, "incompatible types: expected '[int,int...]', " +
+                "found '[int,int,int...]'", 38, 13);
+        BAssertUtil.validateError(negative, i++, "incompatible types: expected '([int,int...]|[int,int,int...]" +
+                "|int[2])', found '[int,int,int...]'", 45, 13);
+        BAssertUtil.validateError(negative, i, "invalid list binding pattern: expected an array or a tuple, " +
+                        "but found '([int,int...]|[int,int,int...])'", 52, 17);
```

Comment:
This error here is a bit confusing, right? Because `[int,int...]|[int,int,int...]` is a subtype of list? Shouldn't the error be similar to the errors above?